src/nodejs/hooks/sealights.go

```diff
@@ -346,62 +402,140 @@ func (sl *SealightsHook) injectSealights(stager *libbuildpack.Stager) error {
 	}
 }
 
-func containsSealightsService(key string, services interface{}, query string) bool {
-	var serviceName string
+func (sl *SealightsHook) parseVcapServices() {
 
-	if strings.Contains(key, query) {
-		return true
+	if sl.initialized {
+		sl.Log.Debug("already initialized. config won`t be parsed")
+		return
+	} else {
+		sl.initialized = true
 	}
-	val := services.([]interface{})
-	for serviceIndex := range val {
-		service := val[serviceIndex].(map[string]interface{})
-		if v, ok := service["name"]; ok {
-			serviceName = v.(string)
-		}
-		if strings.Contains(serviceName, query) {
-			return true
+
+	var vcapServices map[string][]struct {
+		Name        string                 `json:"name"`
+		Credentials map[string]interface{} `json:"credentials"`
+	}
+
+	if err := json.Unmarshal([]byte(os.Getenv("VCAP_SERVICES")), &vcapServices); err != nil {
+		sl.Log.Debug("Failed to unmarshal VCAP_SERVICES: %s", err)
+		return
+	}
+
+	for _, services := range vcapServices {
+		for _, service := range services {
+			if !strings.Contains(strings.ToLower(service.Name), "sealights") {
+				continue
+			}
+
+			queryString := func(key string) string {
+				if value, ok := service.Credentials[key].(string); ok {
+					return value
+				}
+				return ""
+			}
+
+			options := &SealightsParameters{
+				Token:          queryString("token"),
+				TokenFile:      queryString("tokenFile"),
```

Comment:
@CubikNeRubik we have to have the `labId`, `buildSessionIdFile` and `buildSessionId` as part of this release.